tests/unit_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,226 @@
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'fake_project_id'
+        self.ticket_number = 'DC_fake'
+        self.pids_project_id = 'fake_pids_project_id'
+        self.pids_dataset_id = 'fake_pids_dataset_id'
+        self.pids_table = 'fake_pids_table'
+        self.pids_dataset_list = [
+            'fake_dataset_1', 'fake_dataset_2', 'fake_dataset_3'
+        ]
+        self.pids_table_list = ['fake_table_1', 'fake_table_2', 'fake_table_3']
+
+        mock_bq_client_patcher = patch(
+            'retraction.retract_deactivated_pids.get_client')
+        self.mock_bq_client = mock_bq_client_patcher.start()
+        self.addCleanup(mock_bq_client_patcher.stop)
+
+    def test_get_pids_table_info(self):
+        for dataset in self.pids_dataset_list:
+            result_df = self.mock_bq_client.query(
+                retract_deactivated_pids.TABLE_INFORMATION_SCHEMA.render(
+                    project=self.project_id, dataset=dataset)).to_dataframe()
+            expected_pids_tables = []
+            for row in result_df.iterrows():
+                column = getattr(row, 'column_name')
+                table = getattr(row, 'table_name')
+                if 'person_id' in column:
+                    expected_pids_tables.append(table)
+            expected_pids_tables_info_df = result_df[
+                result_df['table_name'].isin(expected_pids_tables)]
+
+            returned_result_df = retract_deactivated_pids.get_pids_table_info(
+                self.project_id, dataset, self.mock_bq_client)
+            self.assertEqual(expected_pids_tables_info_df, returned_result_df)
+
+    def test_get_date_info_for_pids_tables(self):
+
+        date_fields_info_df = pd.DataFrame()
+        dataset_obj = self.mock_bq_client.list_datasets(self.project_id)
+        datasets = [d.dataset_id for d in dataset_obj]
+
+        for dataset in datasets:
+            pids_tables_df = retract_deactivated_pids.get_pids_table_info(
+                self.project_id, dataset, self.mock_bq_client)
+
+            # Keep only records with datatype of 'DATE'
+            date_fields_df = pids_tables_df[pids_tables_df['data_type'] ==
+                                            'DATE']
+
+            # Create df to append to, keeping only one record per table
+            df_to_append = pd.DataFrame(columns=[
+                'project_id', 'dataset_id', 'table', 'date_column',
+                'start_date_column', 'end_date_column'
+            ])
+            df_to_append['project_id'] = date_fields_df['table_catalog']
+            df_to_append['dataset_id'] = date_fields_df['table_schema']
+            df_to_append['table'] = date_fields_df['table_name']
+            df_to_append = df_to_append.drop_duplicates()
+
```

Comment:
This test has the same problem as the previous test.  This test duplicates the code you are trying to test to create the expected result.  This test should also fail, but it doesn't because it contains the same errors.  Line 101 is incorrect and the test should fail.

The assertion on line 119 passes because both the expected and result data frames are empty.  I didn't write a patch file for this, but I can help if needed.